### PR TITLE
Add factory functions to python frontend

### PR DESCRIPTION
**Pull request by rdspring1**
- Add `full` nvprim to support factory functions because the full reference uses `empty` and `fill` while we have a full factory function.
- Change `full_like` reference to call `full` to avoid defining another nvprim.
- Enable support for new_zeros to enable `cudnn_batch_norm` decomposition.

cc @ezyang @mruberry @ngimel @Lezcano @fdrocha @peterbell10

### CODE REVIEW

**Comment by mruberry** [comment]
Let's add a test, too! Also added @kevinstephano to review

File: torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -263,6 +263,24 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+                {fd.recordingState(arg())},
+                {fd.recordingState(output())},
+                size,
+                dtype));
+            return output;
+          },
+          py::arg("size"),
+          py::arg("arg"),
+          py::arg("dtype"),
+          py::return_value_policy::reference)
```
Comment (kevinstephano) [inline torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:283]: `full` should be put under the `ops` with other operations like `view` and `squeeze`.

**Review by kevinstephano — CHANGES_REQUESTED** [review]

**Review by kevinstephano — APPROVED** [review]
LGTM.

**Comment by rdspring1** [comment]
@mruberry I added tests to test/test_prims.py, so the PR is ready.

**Review by mruberry — APPROVED** [review]
Stamped!

**Pull request closed without merging**